.github/workflows/check-try-state.yml

```
@@ -24,17 +24,47 @@ jobs:
     steps:
```

First extend matrix with 
```
runtime:
          - { name: astar, endpoint: ASTAR_RUNTIME_ENDPOINT, http: 'https://rpc.astar.network' }
          - { name: shiden, endpoint: SHIDEN_RUNTIME_ENDPOINT, http: 'https://rpc.shiden.astar.network' }
          - { name: shibuya, endpoint: SHIBUYA_RUNTIME_ENDPOINT, http: 'https://rpc.shibuya.astar.network' }
```
and a step before checkout to find the tag:
```suggestion
    steps:
       - run: |
           SPEC_VERSION=$(curl -s -L ${{ matrix.runtime.http }}  \
           --header 'Content-Type: application/json' \
           --data '{
           "jsonrpc": "2.0",
           "method": "state_getRuntimeVersion",
           "params": [],
           "id": 1
           }' | jq -r .result.specVersion)
           echo "TAG=runtime-$SPEC_VERSION" >> $GITHUB_ENV
```
finally checkout at ref:
```
   uses: actions/checkout@v4
        with:
          ref: ${{ env.TAG }}
          fetch-depth: 0
```